Merge Shopper Insights into Main for release

tdchow opened the pull request:
### Summary of changes

 - Merge Shopper Insights and Merchant Passed Email features into main for release

### Checklist

 - [ ] Added a changelog entry
 - [ ] Relevant test coverage

### Authors
> List GitHub usernames for everyone who contributed to this pull request.

tdchow commented on ShopperInsights/src/main/java/com/braintreepayments/api/ShopperInsightsClient.kt:
```diff
@@ -0,0 +1,175 @@
+        }
+
+        // TODO: get correct merchant ID from SDK
+        val merchantId = "MXSJ4F5BADVNS"
```
This hardcoded merchant ID needs to get replaced.

tdchow replied:
Hey @agedd - are we dynamically setting merchant ID on iOS?

jaxdesmarais replied:
It doesn't look like we have the merchant ID at all on iOS: https://github.com/braintree/braintree_ios/blob/payment-insights-feature/Sources/BraintreeShopperInsights/BTEligiblePaymentsRequest.swift

tdchow replied:
Oh interesting - I think the backend is determining which merchant based on the auth fingerprint. I'll try removing the merchant ID from the request.

agedd replied:
hi @tdchow! yep we removed it [here](https://github.com/braintree/braintree_ios/pull/1195) since it's now determined by the auth fingerprint.

tdchow replied:
Thanks @agedd! I'll make the same change on Android.

jaxdesmarais commented on CHANGELOG.md:
```diff
@@ -1,5 +1,12 @@
+
+* PayPal
+    * Add optional property `PayPalCheckoutRequest.setUserAuthenticationEmail()`
+* Add ShopperInsights module (BETA)
```
```suggestion
* ShopperInsights (BETA)
```
We should also probably add a description here of the new methods/how a merchant would use this. We can do something similar to what we did for PayPalMessaging below

tdchow replied:
Sarah mentioned that this closely matches what iOS has in the changelog: https://github.com/braintree/braintree_android/pull/1006#discussion_r1608747071


Maybe adding this line back is enough?
```
  * Add `ShopperInsightsClient.getRecommendedPaymentMethods()` for returning recommendations based on the buyer
```

jaxdesmarais replied:
Gotcha, yeah I get removing some of the complexity but I think we'd want to call out the public interfaces that we'd want merchants to use. Otherwise we just drop this feature with no details. Curious what @scannillo thinks as well.

tdchow replied:
Makes sense! I'll add that bullet back.

tdchow replied:
Added the bullet back in https://github.com/braintree/braintree_android/pull/1024

jaxdesmarais commented on Demo/src/main/java/com/braintreepayments/demo/Settings.java:
```diff
@@ -27,6 +27,9 @@ public class Settings {
     static final String LOCAL_PAYMENTS_TOKENIZATION_KEY = "sandbox_f252zhq7_hh4cpc39zq4rgjcg";
 
+    private static final String XO_SANDBOX_TOKENIZATION_KEY = "sandbox_rz48bqvw_jcyycfw6f9j4nj9c";
+    private static final String XO_PRODUCTION_TOKENIZATION_KEY = "";
```
Should this be empty?

tdchow replied:
🤔 Probably not. I'll look into this one.

tdchow replied:
I tested the default prod key, which shows the new checkout experience. I'll remove this empty key.

tdchow replied:
Removed in https://github.com/braintree/braintree_android/pull/1024

jaxdesmarais commented on Demo/src/main/java/com/braintreepayments/demo/Settings.java:
```diff
@@ -27,6 +27,9 @@ public class Settings {
 
     static final String LOCAL_PAYMENTS_TOKENIZATION_KEY = "sandbox_f252zhq7_hh4cpc39zq4rgjcg";
 
+    private static final String XO_SANDBOX_TOKENIZATION_KEY = "sandbox_rz48bqvw_jcyycfw6f9j4nj9c";
```
Are we able to use our default sandbox key that works with our sample merchant server?

tdchow replied:
I think we needed to use a specific sandbox key initially for testing. @agedd - does iOS use the default sandbox key for the MPE flow?

agedd replied:
hmm i think i've only ever tested MPE w/ the clientToken auth flow, but i think these changes may be related to https://github.com/braintree/braintree_android/pull/996 and https://github.com/braintree/braintree_ios/pull/1298 where it looks like we're adding this new key to test the web checkout flow 👀

tdchow replied:
Ah that's right! I'm not sure if our default sandbox key triggers the new web checkout flow. We should be safe to merge this in as is since the keys are determined by a setting preference.

jaxdesmarais replied:
Did we check our default sandbox key? At one point in time I think it was supposed to be ramped for this flow but don't know if that was ever completed. It's fine as is if this is the only way to see that flow in sandbox.

tdchow replied:
I just did a quick test, I'm seeing the old flow for one-time checkout for the default sandbox key. So we'll need to keep this in.

jaxdesmarais commented on Demo/src/main/res/xml/network_security_config.xml:
```diff
@@ -0,0 +1,10 @@
+<network-security-config>
+    <debug-overrides>
+        <trust-anchors>
+            <!-- Trust user added CAs while debuggable only -->
```
Do we still need this?

tdchow replied:
Yep I believe this is still needed. @saperi22 was having cert issues on his emulator and needed to have the demo app trust the user added CA.

jaxdesmarais commented on ShopperInsights/src/main/java/com/braintreepayments/api/ShopperInsightsApi.kt:
```diff
@@ -0,0 +1,14 @@
+internal class ShopperInsightsApi(
+    private val eligiblePaymentsApi: EligiblePaymentsApi
+) {
+
```
```suggestion
```

jaxdesmarais commented on build.gradle:
```diff
@@ -13,7 +13,7 @@ buildscript {
             "androidxTest"   : "1.4.0",
             "room"           : "2.4.0",
-            "playServices"   : "19.2.1",
+            "playServices"   : "19.3.0",
```
Do we need to call out any of the version bumps in this file or are they all internal?

tdchow replied:
It doesn't look like we have been calling out any bumps in internal libraries.

tdchow commented:
Converting to a draft PR to prevent it from being merged and adding the do not merge label. I'll be addressing the PR comments in a separate PR into the feature branch.

tdchow commented:
@jaxdesmarais - the PR should be ready for a re-review!

jaxdesmarais reviewed (approved):
🚀
